Bug: `Generate Spicepod JSON schema` action is failing with '`Arc<str>: Serialize` is not satisfied'

## Describe the bug

Run
>cargo build --manifest-path tools/spicepodschema/Cargo.toml

Observe:

```shell
    Skipping git submodule `https://github.com/duckdb/duckdb` due to update strategy in .gitmodules
   Compiling spicepod v0.18.1-beta (/Users/sg/spice/spiceai/crates/spicepod)
error[E0277]: the trait bound `Arc<str>: Serialize` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:103:24
     |
103  | #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
     |                        ^^^^^^^^^ the trait `Serialize` is not implemented for `Arc<str>`
...
109  |     #[serde(default = "default_truncated")]
     |     - required by a bound introduced by this call
     |
     = note: for local types consider adding `#[derive(serde::Serialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Serialize`:
               &'a T
               &'a mut T
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 183 others
note: required by a bound in `_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/ser/mod.rs:1867:21
     |
1865 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1866 |     where
1867 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Arc<str>: Deserialize<'_>` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:110:26
     |
110  |     pub captured_output: Arc<str>,
     |                          ^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Arc<str>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 202 others
note: required by a bound in `next_element`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/de/mod.rs:1732:12
     |
1730 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1731 |     where
1732 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `Arc<str>: Deserialize<'_>` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:112:27
     |
112  |     pub retention_period: Arc<str>,
     |                           ^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Arc<str>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 202 others
note: required by a bound in `next_element`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/de/mod.rs:1732:12
     |
1730 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1731 |     where
1732 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `Arc<str>: Deserialize<'_>` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:114:35
     |
114  |     pub retention_check_interval: Arc<str>,
     |                                   ^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Arc<str>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 202 others
note: required by a bound in `next_element`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/de/mod.rs:1732:12
     |
1730 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1731 |     where
1732 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `Arc<str>: Deserialize<'_>` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:110:26
     |
110  |     pub captured_output: Arc<str>,
     |                          ^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Arc<str>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 202 others
note: required by a bound in `next_value`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/de/mod.rs:1871:12
     |
1869 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1870 |     where
1871 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

error[E0277]: the trait bound `Arc<str>: Deserialize<'_>` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:112:27
     |
112  |     pub retention_period: Arc<str>,
     |                           ^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Arc<str>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 202 others
note: required by a bound in `next_value`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/de/mod.rs:1871:12
     |
1869 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1870 |     where
1871 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

error[E0277]: the trait bound `Arc<str>: Deserialize<'_>` is not satisfied
    --> crates/spicepod/src/component/runtime.rs:114:35
     |
114  |     pub retention_check_interval: Arc<str>,
     |                                   ^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Arc<str>`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `Arc<str>` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T,)
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 202 others
note: required by a bound in `next_value`
    --> /Users/sg/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.210/src/de/mod.rs:1871:12
     |
1869 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1870 |     where
1871 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `spicepod` (lib) due to 7 previous errors
```

--- a/crates/spicepod/src/component/runtime.rs
+++ b/crates/spicepod/src/component/runtime.rs
@@ -344,16 +344,22 @@ pub struct TaskHistory {
     #[serde(default = "default_true")]
     pub enabled: bool,
     #[serde(default = "default_none")]
+    #[cfg_attr(feature = "schemars", schemars(with = "String"))]
     pub captured_output: Arc<str>,
     #[serde(default = "default_retention_period")]
+    #[cfg_attr(feature = "schemars", schemars(with = "String"))]
     pub retention_period: Arc<str>,
     #[serde(default = "default_retention_check_interval")]
+    #[cfg_attr(feature = "schemars", schemars(with = "String"))]
     pub retention_check_interval: Arc<str>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[cfg_attr(feature = "schemars", schemars(with = "Option<String>"))]
     pub min_sql_duration: Option<Arc<str>>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[cfg_attr(feature = "schemars", schemars(with = "Option<String>"))]
     pub captured_plan: Option<Arc<str>>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[cfg_attr(feature = "schemars", schemars(with = "Option<String>"))]
     pub min_plan_duration: Option<Arc<str>>,
 }
 

